airflow/utils/process_utils.py

```diff
@@ -55,7 +55,12 @@ def reap_process_group(
     sig (SIGTERM) to the process group of pid. If any process is alive after timeout
     a SIGKILL will be send.
 
-    :param pgid: process group id to kill
+    :param process_group_id: process group id to kill. This process group id is retrieved earlier from
+           the process id of the process that started that process group. The process that wants to
+           create the group should run `os.setpgid(0, 0)` as the first command it executes which will
+           set group id = process_id. Effectively the process that is the "root" of the group has
+           pid = gid  and all other processes in the group have different pids but the same gid (equal
+           the pid of the root process)
```

Comment:
```suggestion
    :param process_group_id: process group id to kill. This process group id is retrieved earlier from
           the process id of the process that started that process group.
           
           The process that wants to
           create the group should run `os.setpgid(0, 0)` as the first command it executes which will
           set group id = process_id. Effectively the process that is the "root" of the group has
           pid = gid  and all other processes in the group have different pids but the same gid (equal
           the pid of the root process)
```

I'm also not quite sure what you mean by "This process group id is retrieved earlier from the process id of the process that started that process group"

Reply:
I meant what you explained better!